src/typechecker.erl

```diff
@@ -2648,27 +2647,41 @@ update_map_type({type, _, Ty, Arg}, AssocTys)
     when Ty == map, Arg == any;
 	 Ty == any, Arg == []
 	 ->
-    type(map, [{type, P, Assoc, [Key, ValueType]}
-	       || {type, P, Assoc, Key, ValueType} <- AssocTys ]);
+    type(map, [{type, P, map_field_exact, [Key, ValueType]}
+	       || {type, P, _Assoc, Key, ValueType} <- AssocTys ]
+         %% the original type could have any keys
+         %% so we also need to include and optional any() => any()
+         %% in the updated map type
+         %% (map key types can be overlapping, precedence is left to right,
+         %%  so let's append it as the last entry)
+         ++ [type(map_field_assoc, [type(any), type(any)])]
```

Comment:
because of this for example
```
%% When updating map() it can still have any other optional key
-spec h(map()) -> #{a := any(), b => any()}.
h(Ctx) ->
    Ctx2 = Ctx#{a => 5},
    Ctx2.
```
`map()` is basically an alias for `#{any() => any()}`
otoh `#{Tk := Tv}` means this map must have at least one key, and all keys must have the type `Tk`

Reply:
Ok, thanks